WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -729,105 +730,58 @@ private extension StoreCreationCoordinator {
 
     @MainActor
     func waitForSiteToBecomeJetpackSite(from navigationController: UINavigationController, siteID: Int64, expectedStoreName: String) {
-        /// Free trial sites need more waiting time that regular sites.
-        ///
-        siteIDFromStoreCreation = siteID
-
-        /// Determines if the `.siteCreationPropertiesOutOfSync` event has already been tracked.
-        /// Needed because we should only track this event once per store creation process.
-        ///
-        var haveTrackedOutOfSyncEvent = false
-
         /// Timestamp when we start observing times. Needed to track the store creating waiting duration.
         ///
         let waitingTimeStart = Date()
 
-        scheduleLocalNotificationWhenStoreIsReady()
-
-        jetpackSiteSubscription = $siteIDFromStoreCreation
-            .compactMap { $0 }
-            .removeDuplicates()
-            .asyncMap { [weak self] siteID -> Site? in
-                guard let self else {
-                    return nil
+        let statusChecker = StoreCreationStatusChecker(isFreeTrialCreation: isFreeTrialCreation, stores: stores)
+        self.statusChecker = statusChecker
+        jetpackSiteSubscription = statusChecker.waitForSiteToBeReady(siteID: siteID)
+            .handleEvents(receiveCompletion: { [weak self] completion in
+                guard let self, case .failure = completion else {
+                    return
                 }
-                // Waits some seconds before syncing sites every time.
-                try await Task.sleep(nanoseconds: UInt64(self.jetpackCheckRetryInterval * 1_000_000_000))
-                return try await self.syncSite(siteID: siteID, expectedStoreName: expectedStoreName, haveTrackedOutOfSyncEvent: haveTrackedOutOfSyncEvent)
-            }
-            .receive(on: DispatchQueue.main)
-            .handleEvents(receiveCompletion: { [weak self] output in
-                guard let self else { return }
                 self.storeCreationProgressViewModel?.incrementProgress()
-
-                // Track when a properties out of sync event occur, but only track it once.
-                if self.isPropertiesOutOfSyncError(output) && !haveTrackedOutOfSyncEvent {
-                    self.analytics.track(event: .StoreCreation.siteCreationPropertiesOutOfSync())
-                    haveTrackedOutOfSyncEvent = true
-                }
             })
-            // Retries 10 times with some seconds pause in between to wait for the newly created site to be available as a Jetpack site
-            // in the WPCOM `/me/sites` response.
+            // Retries 10 times with some seconds pause in between to wait for the newly created site to be available as a Jetpack/Woo site.
             .retry(10)
-            .replaceError(with: nil)
-            .sink { [weak self] site in
+            .sink { [weak self] completion in
```

Comment:
Super nit: Should we use explicit params instead of trailing closure to make this easier to understand? I didn't know of the `receiveCompletion` param and had to look up the documentation to understand this part 😅 

```suggestion
            .sink (receiveCompletion: { [weak self] completion in
```

Reply:
Yea it's not the most intuitive, updated in https://github.com/woocommerce/woocommerce-ios/pull/9767/commits/4cb72a6093a7336f2325dcb332867c9c44568658